src/Maestro/DependencyUpdateErrorProcessor/DependencyUpdateErrorProcessor.cs

```diff
@@ -0,0 +1,227 @@
+                                previousTransactionDateTime
+                                );
+                            await tx.CommitAsync();
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        Logger.LogError(ex, "Processing update dependency messages");
+                    }
+                }
+            }
+            else
+            {
+                Logger.LogInformation("Dependency Update Error processor is disabled because no App Configuration was available.");
+            }
+        }
+
+        public async Task<DateTimeOffset> CheckForError(DateTimeOffset previousTransactionDateTime , string whereToCreateIssue)
+        {
+            List<RepositoryBranchUpdateHistoryEntry> list = (from repo in Context.RepositoryBranchUpdateHistory
+                                                             where repo.Success == false
+                                                             where repo.Timestamp > previousTransactionDateTime.UtcDateTime
+                                                             orderby repo.Timestamp ascending
+                                                             select repo).ToList();
+            if (!list.Any())
+            {
+                return previousTransactionDateTime;
+            }
+
+            Logger.LogInformation($"Going to create the github issue");
+            try
+            {
+                await CreateGitHubIssueAsync(previousTransactionDateTime.UtcDateTime,  list[0].Repository, list[0].Branch, 
+                    list[0].Action, list[0].ErrorMessage, list[0].Method , whereToCreateIssue);
+            }
+            catch (Exception ex)
+            {
+                Logger.LogError(ex, "Could not create a github issue for the error: " + list[0].ErrorMessage);
+            }
+            previousTransactionDateTime = list[0].Timestamp;
+            return previousTransactionDateTime;
+        }
+
+        public async Task<bool> SearchCreatedIssueAsync(DateTime transactionDateTime , string errorMessage)
+        {
+            var query = new StringBuilder();
+            HttpClient client = new HttpClient
+            {
+                BaseAddress = new Uri("https://api.github.com")
+            };
+            client.DefaultRequestHeaders.Add("User-Agent", @"Mozilla/5.0 (Windows NT 10; Win64; x64; rv:60.0) Gecko/20100101 Firefox/60.0");
+
+            query.Append($"is:issue+is:open+{"Error Message: " + errorMessage}in:title&order=desc");
+            JObject responseContent;
+            using (HttpResponseMessage response = await client.GetAsync(
+                $"search/issues?q={query}"))
+            {
+                responseContent = JObject.Parse(await response.Content.ReadAsStringAsync());
+            }
+            JArray items = JArray.Parse(responseContent["items"].ToString());
+            if (!items.IsNullOrEmpty())
```

Comment:
This could be flipped, so it's
```csharp
if (items.IsNullOrEmpty())
    return true;
```

That way I don't need to scroll all the way down to see what happens if it IS null or empty, I can see really quick "oh, if it's empty, we call that good and return", and can then forget about that case.  Right now I need to keep that context of "all this code is only if this if", so that when I get to the end of this if, I can pop that off my brain stack, and now have to deal with what might happen for both branches of the if.  (Turns out, my poor brain wasted all that effort, because it's just a bunch of unconditional ifs at the end... but I need then scroll up to see which if statements end where causing which values to get returned).

Also, it will make the rest of the code less indented, which is always easier for readability.